### PR TITLE
CON-2653 Added reset for all answers in application section

### PR DESCRIPTION
**Pull request by ChrisJWoodcock**

### CODE REVIEW

File: src/SFA.DAS.QnA.Application/Commands/ResetPageAnswers/ResetPageAnswersBySectionNoHandler.cs [hunk 1]
```diff
@@ -20,6 +18,7 @@ public ResetPageAnswersBySectionNoHandler(QnaDataContext dataContext, INotRequir
 
         public async Task<HandlerResponse<ResetPageAnswersResponse>> Handle(ResetPageAnswersBySectionNoRequest request, CancellationToken cancellationToken)
         {
+            var application = await _dataContext.Applications.SingleOrDefaultAsync(app => app.Id == request.ApplicationId, cancellationToken);
```
Comment (martin-lush) [inline src/SFA.DAS.QnA.Application/Commands/ResetPageAnswers/ResetPageAnswersBySectionNoHandler.cs:21]: I'd move this back to being below the validationErrorResponse  so you're not wasting a database call if validation fails

**Review by MarkFCain — DISMISSED** [review]

File: src/SFA.DAS.QnA.Application/Commands/ResetPageAnswers/ResetPageAnswersHandler.cs [hunk 2]
```diff
@@ -19,6 +17,7 @@ public ResetPageAnswersHandler(QnaDataContext dataContext, INotRequiredProcessor
 
         public async Task<HandlerResponse<ResetPageAnswersResponse>> Handle(ResetPageAnswersRequest request, CancellationToken cancellationToken)
         {
+            var application = await _dataContext.Applications.SingleOrDefaultAsync(app => app.Id == request.ApplicationId, cancellationToken);
```
Comment (martin-lush) [inline src/SFA.DAS.QnA.Application/Commands/ResetPageAnswers/ResetPageAnswersHandler.cs:20]: I'd move this back to being below the validationErrorResponse so you're not wasting a database call if validation fails

File: src/SFA.DAS.QnA.Application/Commands/ResetSectionAnswers/ResetSectionAnswersHandler.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+        public ResetSectionAnswersHandler(QnaDataContext dataContext, IMediator mediator, INotRequiredProcessor notRequiredProcessor, ITagProcessingService tagProcessingService) 
+            : base(dataContext, notRequiredProcessor, tagProcessingService, null)
+        {
+            _mediator = mediator;
+        }
+
+        public async Task<HandlerResponse<ResetSectionAnswersResponse>> Handle(ResetSectionAnswersRequest request, CancellationToken cancellationToken)
+        {
+            var application = await _dataContext.Applications.SingleOrDefaultAsync(app => app.Id == request.ApplicationId, cancellationToken);
```
Comment (martin-lush) [inline src/SFA.DAS.QnA.Application/Commands/ResetSectionAnswers/ResetSectionAnswersHandler.cs:27]: I'd move this to being below the validationErrorResponse so you're not wasting a database call if validation fails

**Review by martin-lush — CHANGES_REQUESTED** [review]
3 minor comments regarding performance. Otherwise can't see any issues

File: src/SFA.DAS.QnA.Application/Commands/ResetPageAnswers/ResetPageAnswersBySectionNoHandler.cs (hunk 1 above)
Comment (ChrisJWoodcock) [reply src/SFA.DAS.QnA.Application/Commands/ResetPageAnswers/ResetPageAnswersBySectionNoHandler.cs]: Done

File: src/SFA.DAS.QnA.Application/Commands/ResetPageAnswers/ResetPageAnswersHandler.cs (hunk 2 above)
Comment (ChrisJWoodcock) [reply src/SFA.DAS.QnA.Application/Commands/ResetPageAnswers/ResetPageAnswersHandler.cs]: Done

File: src/SFA.DAS.QnA.Application/Commands/ResetSectionAnswers/ResetSectionAnswersHandler.cs (hunk 3 above)
Comment (ChrisJWoodcock) [reply src/SFA.DAS.QnA.Application/Commands/ResetSectionAnswers/ResetSectionAnswersHandler.cs]: Done

**Review by martin-lush — APPROVED** [review]

**Pull request merged**